### PR TITLE
IPAM plugin: improve error messages

### LINKED ISSUE #52: Error message for missing IPAM plugins confusing
The error message from the IPAM plugin search is confusing particularly if the IPAM entry is missing from the config but required.

### PR DESCRIPTION
**Pull request by NeilW**
Make it more clear that we failed to find an IPAM plugin.
Check for a missing plugin name and issue a more helpful error.

Fixes #52

### CODE REVIEW

File: pkg/plugin/ipam.go [hunk 1]
```diff
@@ -63,10 +63,13 @@ func ExecAdd(plugin string, netconf []byte) (*Result, error) {
 	if os.Getenv("CNI_COMMAND") != "ADD" {
 		return nil, fmt.Errorf("CNI_COMMAND is not ADD")
 	}
+	if plugin == "" {
+		return nil, fmt.Errorf(`Name of IPAM plugin is missing. Specify a "type" field in the "ipam" section`)
+        }
```
Comment (steveeJ) [inline pkg/plugin/ipam.go]: The formatting is off. Please always use [gofmt](http://golang.org/cmd/gofmt/) before contributing code.

File: pkg/plugin/ipam.go [hunk 2]
```diff
@@ -63,10 +63,13 @@ func ExecAdd(plugin string, netconf []byte) (*Result, error) {
 	if os.Getenv("CNI_COMMAND") != "ADD" {
 		return nil, fmt.Errorf("CNI_COMMAND is not ADD")
 	}
+	if plugin == "" {
+		return nil, fmt.Errorf(`Name of IPAM plugin is missing. Specify a "type" field in the "ipam" section`)
```
Comment (steveeJ) [inline pkg/plugin/ipam.go]: Better yet: *Please* specify a "type" field..

Comment (jonboulle) [inline pkg/plugin/ipam.go]: nit, please start the error message lower-case

File: pkg/plugin/ipam.go (hunk 1 above)
Comment (jonboulle) [inline pkg/plugin/ipam.go]: Created #55 to catch these automatically in future.

**Comment by NeilW** [comment]
I've updated the message. Hope that's more to standard.

**Comment by jonboulle** [comment]
@NeilW the patch LGTM, thanks, but one minor thing: could you please rebase so we can lose that superfluous merge commit? (355ea08). then we're good to go - thanks!

**Comment by NeilW** [comment]
heh. I was going to do that but didn't want to be seen as rewriting
history.

Eventually I'll make the right call here :-)



On 8 September 2015 at 18:41, Jonathan Boulle <notifications@github.com>
wrote:

> @NeilW <https://github.com/NeilW> the patch LGTM, thanks, but one minor
> thing: could you please rebase so we can lose that superfluous merge
> commit? (355ea08
> <https://github.com/appc/cni/commit/355ea088f050536b34b191e53ce19eb22f9185f1>).
> then we're good to go - thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/appc/cni/pull/53#issuecomment-138644972>.
>



-- 
Neil Wilson

**Comment by jonboulle** [comment]
sorry about our pedantry!

**Comment by steveeJ** [comment]
LGTM

**Pull request merged**